src/libsyncengine/jobs/network/upload_session/uploadsessionlog.cpp

```diff
@@ -0,0 +1,100 @@
+#include "libparms/db/parmsdb.h"
+
+namespace KDC {
+
+UploadSessionLog::UploadSessionLog(const SyncPath &filepath, uint64_t nbParalleleThread /*= 1*/)
+    : AbstractUploadSession(filepath, filepath.filename(), nbParalleleThread) {
+    _uploadSessionType = UploadSessionType::LogUpload;
+}
+
+bool UploadSessionLog::runJobInit() {
+    return true;
+}
+
+std::shared_ptr<UploadSessionStartJob> UploadSessionLog::createStartJob() {
+    return std::make_shared<UploadSessionStartJob>(UploadSessionType::LogUpload, getFileName(), getFileSize(), getTotalChunks());
+}
+
+std::shared_ptr<UploadSessionChunkJob> UploadSessionLog::createChunkJob(const std::string &chunckContent, uint64_t chunkNb,
+                                                                        std::streamsize actualChunkSize) {
+    return std::make_shared<UploadSessionChunkJob>(UploadSessionType::LogUpload, getFilePath(), getSessionToken(), chunckContent,
+                                                   chunkNb, actualChunkSize, jobId());
+}
+
+std::shared_ptr<UploadSessionFinishJob> UploadSessionLog::createFinishJob() {
+    SyncTime modtimeIn =
+        std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now()).time_since_epoch().count();
+
+    return std::make_shared<UploadSessionFinishJob>(UploadSessionType::LogUpload, getFilePath(), getSessionToken(),
+                                                    getTotalChunkHash(), getTotalChunks(), modtimeIn);
+}
+
+std::shared_ptr<UploadSessionCancelJob> UploadSessionLog::createCancelJob() {
+    return std::make_shared<UploadSessionCancelJob>(UploadSessionType::LogUpload, getSessionToken());
+}
+
+bool UploadSessionLog::prepareChunkJob(const std::shared_ptr<UploadSessionChunkJob> &chunkJob) {
+    return true;
+}
+
+bool UploadSessionLog::handleStartJobResult(const std::shared_ptr<UploadSessionStartJob> &StartJob, std::string uploadToken) {
+    AppStateValue appStateValue = ""; // 
+    if (bool found = false; !ParmsDb::instance()->selectAppState(AppStateKey::LogUploadToken, appStateValue, found) || !found) {
+        LOG_WARN(getLogger(), "Error in ParmsDb::selectAppState");
+    }
+    const std::string logUploadToken = std::get<std::string>(appStateValue);
+    if (!logUploadToken.empty()) {
+        UploadSessionCancelJob cancelJob(UploadSessionType::LogUpload, logUploadToken);
+        const ExitCode exitCode = cancelJob.runSynchronously();
+        if (exitCode != ExitCodeOk) {
+            LOG_WARN(getLogger(), "Error in UploadSessionCancelJob::runSynchronously : " << exitCode);
+        } else {
+            LOG_INFO(getLogger(), "Previous Log upload api call cancelled");
+            bool found = true;
+            if (!ParmsDb::instance()->updateAppState(AppStateKey::LogUploadToken, std::string(), found) || !found) {
+                LOG_WARN(getLogger(), "Error in ParmsDb::updateAppState");
+            }
+        }
+    }
+
+    AppStateValue value = uploadToken;
```

Comment:
Why is useful to create a `value` variable of type `AppStateValue`?

Reply:
In this case, it is not useful, I replace it in commit: b827ff08